pkg/config/remote/service/service.go

```diff
@@ -925,3 +951,180 @@ func enforceCanonicalJSON(raw []byte) ([]byte, error) {
 
 	return canonical, nil
 }
+
+// HTTPClient fetches Remote Configurations from an HTTP(s)-based backend
+type HTTPClient struct {
+	Service
+	lastUpdate time.Time
+	cdnUptane  cdnUptaneClient
+}
+
+// NewHTTPClient creates a new HTTPClient that can be used to fetch Remote Configurations from an HTTP(s)-based backend
+// It uses a local db to cache the fetched configurations. Only one HTTPClient should be created per agent.
+// An HTTPClient must be closed via HTTPClient.Close() before creating a new one.
+func NewHTTPClient(runPath, site, apiKey, agentVersion string) (*HTTPClient, error) {
+	dbPath := path.Join(runPath, "remote-config-cdn.db")
+	db, err := openCacheDB(dbPath, agentVersion, apiKey)
+	if err != nil {
+		return nil, err
+	}
+
+	uptaneCDNClient, err := uptane.NewCDNClient(db, site, apiKey)
+	if err != nil {
+		return nil, err
+	}
+
+	return &HTTPClient{
+		Service: Service{
+			rcType: "CDN",
+			uptane: uptaneCDNClient,
+			db:     db,
+		},
+		cdnUptane: uptaneCDNClient,
+	}, nil
+}
+
+// Close closes the HTTPClient and cleans up any resources. Close must be called
+// before any other HTTPClients are instantiated via NewHTTPClient
+func (c *HTTPClient) Close() error {
+	return c.db.Close()
+}
+
+// GetCDNConfigUpdate returns any updated configs. If multiple requests have been made
+// in a short amount of time, a cached response is returned. If RC has been disabled,
+// an error is returned. If there is no update (the targets version is up-to-date) nil
+// is returned for both the update and error.
+func (c *HTTPClient) GetCDNConfigUpdate(
+	ctx context.Context,
+	products []string,
+	currentTargetsVersion, currentRootVersion uint64,
+	cachedTargetFiles []*pbgo.TargetFileMeta,
+) (*state.Update, error) {
+	var err error
+	span, ctx := tracer.StartSpanFromContext(ctx, "HTTPClient.GetCDNConfigUpdate")
+	defer span.Finish(tracer.WithError(err))
+	if !c.shouldUpdate() {
+		span.SetTag("use_cache", true)
+		return c.getUpdate(ctx, products, currentTargetsVersion, currentRootVersion, cachedTargetFiles)
+	}
+
+	err = c.update(ctx)
+	if err != nil {
+		span.SetTag("cache_update_error", true)
+		_ = log.Warn(fmt.Sprintf("Error updating CDN config repo: %v", err))
```

Comment:
Let's error out here, if it's a real TUF issue we'll have no config and evict everything

Reply:
Talked offline. Takeaway is that:
- this will stay the same, and continue to mimic the behavior of the core agent cache update behavior (i.e. "rollback" to most recent stable cache state in the event of an update failure)
- there is a separate issue that will be addressed separately from this PR, where the installer daemon and cmd both have different caches, meaning if one succeeds and the following fails to do updates, an operation could have unexpected behaviors that wouldn't be present if a single client were performing all the operations.